fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/AbstractMaterializedViewAggregateRule.java

```diff
@@ -230,57 +230,56 @@ protected Plan rewriteQueryByView(MatchMode matchMode,
-                                    queryFunctionShuttled, mvExprToMvScanExprQueryBased));
-                    return null;
-                }
-                finalOutputExpressions.add(new Alias(rollupedExpression));
-            } else {
-                // if group by expression, try to rewrite group by expression
-                Expression queryGroupShuttledExpr = ExpressionUtils.shuttleExpressionWithLineage(
-                        topExpression, queryTopPlan, queryStructInfo.getTableBitSet());
-                AggregateExpressionRewriteContext context = new AggregateExpressionRewriteContext(true,
-                        mvExprToMvScanExprQueryBased, queryTopPlan, queryStructInfo.getTableBitSet());
-                // group by expression maybe group by a + b, so we need expression rewriter
-                Expression rewrittenGroupByExpression = queryGroupShuttledExpr.accept(AGGREGATE_EXPRESSION_REWRITER,
-                        context);
-                if (!context.isValid()) {
-                    // group expr can not rewrite by view
-                    materializationContext.recordFailReason(queryStructInfo,
-                            "View dimensions doesn't not cover the query dimensions",
-                            () -> String.format("mvExprToMvScanExprQueryBased is %s,\n queryGroupShuttledExpr is %s",
-                                    mvExprToMvScanExprQueryBased, queryGroupShuttledExpr));
-                    return null;
-                }
-                NamedExpression groupByExpression = rewrittenGroupByExpression instanceof NamedExpression
-                        ? (NamedExpression) rewrittenGroupByExpression : new Alias(rewrittenGroupByExpression);
-                finalOutputExpressions.add(groupByExpression);
-                finalGroupExpressions.add(groupByExpression);
+            Expression queryFunctionShuttled = ExpressionUtils.shuttleExpressionWithLineage(
+                    topExpression,
+                    queryTopPlan,
+                    queryStructInfo.getTableBitSet());
+            AggregateExpressionRewriteContext context = new AggregateExpressionRewriteContext(
+                    false, mvExprToMvScanExprQueryBased, queryTopPlan, queryStructInfo.getTableBitSet());
+            // queryFunctionShuttled maybe sum(column) + count(*), so need to use expression rewriter
+            Expression rollupedExpression = queryFunctionShuttled.accept(AGGREGATE_EXPRESSION_REWRITER,
+                    context);
+            if (!context.isValid()) {
+                materializationContext.recordFailReason(queryStructInfo,
+                        "Query function roll up fail",
+                        () -> String.format("queryFunctionShuttled = %s,\n mvExprToMvScanExprQueryBased = %s",
+                                queryFunctionShuttled, mvExprToMvScanExprQueryBased));
+                return null;
             }
+            finalOutputExpressions.add(new Alias(rollupedExpression));
```

Comment:
final output should contains both aggregate function and group by dimension， this may lost group by dimension out put and change original agg output

Reply:
> final output should contains both aggregate function and group by dimension， this may lost group by dimension out put and change original agg output

I think maybe line 261 does the job you mentioned.
This PR does not modify the logic in rewriteQueryByView().
The key change is line 428 in topPlanSplitToGroupAndFunction() that prevent the loss of groupby in child agg for rules like MaterializedViewProjectAggregateRule.

Reply:
I get your point, the change may change the original agg output order. 
such as query is as following:
```
SELECT sum(o_totalprice), o_orderpriority
FROM orders
GROUP BY o_orderstatus, o_orderpriority;
```
after mv rewreite will be 
```
SELECT o_orderpriority, sum(o_totalprice)
FROM orders
GROUP BY o_orderstatus, o_orderpriority;
```

Maybe we should add another check as following
WDYT?
```
        // if top plan doesn't use group by but bottom agg has group by, should also check
        if (!queryGroupByExpressions.isEmpty() && queryTopPlanGroupBySet.isEmpty()) {
            // todo check as your code in line 242 to 262
        }
```

Reply:
Thx, I will try